Fabric.Identity.API/scripts/Register-Identity-IdPSS.ps1

```diff
@@ -0,0 +1,47 @@
+param(
+    [ValidateScript({
+        if (!(Test-Path $_)) {
+            throw "Path $_ does not exist. Please enter valid path to the install.config."
+        }
+        if (!(Test-Path $_ -PathType Leaf)) {
+            throw "Path $_ is not a file. Please enter a valid path to the install.config."
+        }
+        return $true
+    })] 
+    [string] $installConfigPath = "$PSScriptRoot\install.config",
+    [Parameter(Mandatory=$true)]
+    [string[]] $tenants,
+    [Parameter(Mandatory=$true)]
+    [string[]] $replyUrls
+)
+
+Import-Module -Name .\Install-IdPSS-Utilities.psm1 -Force
+
+# Import Fabric Install Utilities
+$fabricInstallUtilities = ".\Fabric-Install-Utilities.psm1"
+if (!(Test-Path $fabricInstallUtilities -PathType Leaf)) {
+    Write-DosMessage -Level "Warning" -Message "Could not find fabric install utilities. Manually downloading and installing"
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/InstallScripts/master/common/Fabric-Install-Utilities.psm1 -Headers @{"Cache-Control" = "no-cache"} -OutFile $fabricInstallUtilities
+}
+Import-Module -Name $fabricInstallUtilities -Force
+
+# TODO: Differentiate between idpss and identity application for extra/different permissions
+if($null -ne $tenants) {
+    foreach($tenant in $tenants) {
+        Write-Host "Enter credentials for specified tenant $tenant"
+        $credential = Get-Credential
+
+        Connect-AzureADTenant -tenantId $tenant -credential $credential
+
+        # TODO: This will potentially redirect to an application that may not be installed yet?
+        $app = New-FabricAzureADApplication -appName 'Identity Provider Search Service' -replyUrls $replyUrls
+        $clientId = $app.AppId
+        $clientSecret = Get-FabricAzureADSecret -objectId $app.ObjectId
+
+        Disconnect-AzureAD
+        Add-InstallationTenantSettings -configSection "common" -tenantId $tenant -clientSecret $clientSecret -clientId $clientId -installConfigPath $installConfigPath
```

Comment:
Putting into common directly, should I make a IdPSS config section to go along or is this fine?

Reply:
I'm thinking we put all IdPSS config and Identity config into the "identity" config section and treat them as one logical application, i.e. they get installed with the same script, and their config lives in the same place in install.config. Of course we would need to spread that config across two web.configs, but to the end user I want the two apps to feel like one.

Reply:
do we need to write the secret to the config file if we re-create each time?

Reply:
Yes, this script registers only, so the install script will need to grab these values somewhere in order to move them into the proper web.configs.  The reason for splitting these two out, is that the registration can't really be automated at the moment since we need human interaction to give admin consent to an application